src/server/scripts/Kalimdor/DireMaul/instance_dire_maul.cpp

```diff
@@ -60,8 +78,205 @@ class instance_dire_maul : public InstanceMapScript
+
+            for (uint8 i = 0; i < 5; ++i)
+            {
+                bool _mobAlive = false;
+                go = instance->GetGameObject(go_crystals[i]);
+                if (!go)
+                    continue;
+
+                if (go->GetGoState() == GO_STATE_READY)
+                {
+                    for (uint8 j = 0; j < 4; ++j)
+                    {
+                        mob = instance->GetCreature(crystalcreaturelist[i][j]);
+                        if (mob && mob->IsAlive())
+                            _mobAlive = true;
+                    }
+                }
+
+                if (!_mobAlive && go->GetGoState() == GO_STATE_READY) // if all stored creatures are dead and go state is ready
+                {
+                    HandleGameObject(ObjectGuid::Empty, false, go);
+
+                    switch (go->GetEntry())
+                    {
+                        case GO_CRYSTAL_01:
+                            SetBossState(DATA_CRYSTAL_01, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_02:
+                            SetBossState(DATA_CRYSTAL_02, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_03:
+                            SetBossState(DATA_CRYSTAL_03, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_04:
+                            SetBossState(DATA_CRYSTAL_04, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_05:
+                            SetBossState(DATA_CRYSTAL_05, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+
+            if (GetBossState(DATA_CRYSTAL_01) == DONE && GetBossState(DATA_CRYSTAL_02) == DONE && GetBossState(DATA_CRYSTAL_03) == DONE &&
+                GetBossState(DATA_CRYSTAL_04) == DONE && GetBossState(DATA_CRYSTAL_05) == DONE)
+            {
+                // if all crystals are done, we set encounter forcefield to done
+                SetBossState(DATA_FORCEFIELD, DONE);
+                // activate forcefield to make it disappear
+                if (GameObject* ffield = instance->GetGameObject(forcefield))
+                    ffield->SetGoState(GO_STATE_ACTIVE);
+                // remove previously set non attackable flag
+                if (Creature* Immo = instance->GetCreature(immoGUID))
+                    Immo->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+            }
+        }
+
+protected:
+        InstanceScript* _instance;
+        EventMap _events;
+        ObjectGuid go_crystals[5];
```

Comment:
_goCrystals pls and go with std::array<ObjectGuid, 5> while at it

all private members should go with   _memberName

Reply:
Okay, well, i understand that stuff with _private members.
But that std::array<ObjectGuid, 5> goes beyond my understanding of c++.
Do you have the time to explain what this does compared to what i do and therefore why this should be changed?

Reply:
https://en.cppreference.com/w/cpp/container/array